### DISCUSSION TITLE
Unexpected Login Screen with Hidden Username After SSO Session Timeout

### DISCUSSION THREAD

**Discussion by adilld** [keycloak/keycloak#35814] [Q&A]
Hello,

I am encountering an issue with SSO authentication using Microsoft Azure AD. After 20 minutes of inactivity, if I open a new tab, I am redirected to the login page, exactely this URL is being displayed on the browser : 

```perl
https://keycloakUrl...realm/protocol/openid-connect/auth?client_id=my-client-id
&redirect_uri=https....
&state=46237bd4-1077-4baa-9fc3-3fbd0fcb7588
&response_mode=fragment
&response_type=code&scope=openid
&nonce=f1acc6ee-8e6a-4a8e-4ca6-db9711d12ee0
&code_challenge=Fx5n78ACQb-C1uXpjN-YZ7D7G5wC6NO1Pr7Z5ytnUXc
&code_challenge_method=S256  
```

However, the page only displays the password field and social provider buttons, while the username field remains hidden due to the following condition in the form template:

```html
<#if !usernameHidden??>  
```

Additionally, when attempting to enter the keycloak's password (or AD) or clicking "Log in using SSO," I encounter a Bad Request error on the following request:

```bash
GET https://keycloakUrl.../login-actions/post-broker-login?client_id=...&tab_id=....&client_data=...  
```
And the HTML page shows the message: We are sorry, Username/password incorrect

### Context and Configuration:
Previously, the session duration was set to 20 minutes, and I updated it to:
SSO idle timeout: 30 minutes
Token lifespan: 28 minutes
My Angular application handles token refresh every 5 minutes.
The issue occurs on two environments but not in the development environment.
I have compared the realm and client configurations across environments, and they are identical.
Despite these checks, I am unable to identify the root cause of this behavior.

Could you please provide guidance or suggest any troubleshooting steps to resolve this issue?

Thank you for your support!

Environment Details:
Keycloak Version: 26.0.5
Angular Version: 18
Keycloak Angular: 16.1.0
Keycloak JS: 25.0.6
Authentication Provider: Microsoft Azure AD

**Comment by codespearhead** [comment]
Try using the Keycloak JS version that matches your Keycloak server.

If that doesn’t resolve the issue, let us know if it can be reproduced in the latest stable release of Keycloak (currently Keycloak 26.0.7 and Keycloak JS 26.0.7).
